examples/type-level-boolean-programming

```diff
@@ -0,0 +1,38 @@
+/// A list that tracks whether its length is even or odd.
+enum EOList[a: Type, even: Bool] {
+    case EONil
+    case EOCons(a, EOList[a, not even])
+}
+
+/// Returns an empty and even list.
+def mkNilEO(): EOList[a, true] = EOList.EONil
+
+/// Attaches an element to the front of the list.
+def consEO(x: a, list: EOList[a, even]): EOList[a, not even] =
+    EOList.EOCons(x, list)
+
+/// Returns the head of the list.
+def headEO(list: EOList[a, even]): a = match list {
```

Comment:
list -> l (everywhere)

Reply:
I don't understand why you have "even" as the type argument, should it not be "false"?

Reply:
I did that because the functionality shouldn't depend on whether the list is even. The intention is that `even` is just a variable used to determine evenness, whether it's `true` or `false`. Should I only allow for returning the heads of odd lists?